packages/ra-input-rich-text/src/RichTextInput.tsx

```diff
@@ -108,10 +108,19 @@ export const RichTextInput = (props: RichTextInputProps) => {
 
     const { error, invalid, isTouched } = fieldState;
 
+    useEffect(() => {
+        if (!editor) return;
+
+        editor.commands.setContent(field.value, false, {
+            preserveWhitespace: true,
+        });
+    }, [editor, field.value]);
+
     useEffect(() => {
         if (!editor) return;
 
         editor.setOptions({
+            content: field.value,
```

Comment:
did you really mean to keep this line? I don't think it's needed (and it causes a linter warning in the useEffect dependency array)